packages/storage/amplify_storage_s3_dart/test/storage_s3_service/task/s3_upload_task_test.dart

```diff
@@ -1078,14 +1208,48 @@ void main() {
             },
           );
 
-          unawaited(uploadTask.start());
+          final completer1 = Completer<void>();
+          final completer2 = Completer<void>();
+          final completer3 = Completer<void>();
 
+          when(
+            () => uploadPartSmithyOperation1.result,
+          ).thenAnswer((_) async {
+            await completer1.future;
+            throw const CancellationException();
+          });
+          when(
+            () => uploadPartSmithyOperation2.result,
+          ).thenAnswer((_) async {
+            await completer2.future;
+            throw const CancellationException();
+          });
+          when(
+            () => uploadPartSmithyOperation3.result,
+          ).thenAnswer((_) async {
+            await completer3.future;
+            throw const CancellationException();
+          });
+
+          await uploadTask.start();
+
+          // add a manual delay to ensure upload parts are scheduled before
```

Comment:
What happens if this delay is not in place? Would a customer potentially have to do something like this?

Reply:
If the delay is not in place, immediate `cancel` will cancel the task, where the http requests won't be schedule at all, where `SmihtyOperation.cancel` won't be called.